extension/src/popup/views/__tests__/MnemonicPhrase.test.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { configureStore, combineReducers } from "@reduxjs/toolkit";
+import {
+  render,
+  waitFor,
+  screen,
+  fireEvent,
+  act,
+} from "@testing-library/react";
+import { Provider } from "react-redux";
+
+import { reducer as auth } from "popup/ducks/authServices";
+import { ROUTES } from "popup/constants/routes";
+
+import { MnemonicPhrase } from "../MnemonicPhrase";
+import { METRIC_NAMES } from "popup/constants/metricsNames";
+const MNEMONIC = "dummy mnemonic";
+
+const rootReducer = combineReducers({
+  auth,
+});
+const makeDummyStore = (state: any) =>
+  configureStore({
+    reducer: rootReducer,
+    preloadedState: state,
+  });
+
+const { MemoryRouter } = jest.requireActual("react-router-dom");
+const Wrapper: React.FunctionComponent<any> = ({ children, state }) => (
+  <MemoryRouter>
+    <Provider store={makeDummyStore(state)}>{children}</Provider>
+  </MemoryRouter>
+);
+
+jest.mock("popup/helpers/download");
+jest.mock("helpers/metrics");
+jest.mock("popup/helpers/useMnemonicPhrase", () => ({
+  useMnemonicPhrase: () => "dummy mnemonic",
+}));
+jest.mock("react-router-dom", () => {
+  const ReactRouter = jest.requireActual("react-router-dom");
+  return {
+    ...ReactRouter,
+    Redirect: ({ to }: any) => <div>redirect {to}</div>,
+  };
+});
+jest.mock("@shared/api/internal", () => {
+  const { APPLICATION_STATE } = jest.requireActual(
+    "@shared/constants/applicationState",
+  );
+  return {
+    confirmMnemonicPhrase: () =>
+      Promise.resolve({
+        isCorrectPhrase: true,
+        applicationState: APPLICATION_STATE.MNEMONIC_PHRASE_CONFIRMED,
+      }),
+  };
+});
```

Comment:
Some of these mocks we might want to pull out into more reusable modules, or into a `__mocks__` folder, as we write more tests